docs/troubleshooting.md

```diff
@@ -0,0 +1,23 @@
+
+### I cannot join a worker node
+
+In [standalone](usage-overview.md#standalone) mode, check whether the join
+token has expired. If this is the case, attempt to create a new one using
+[JoinTokenRequest](https://docs.k0smotron.io/stable/join-nodes/#join-tokens) In
+[Cluster API](cluster-api.md), check the logs of your infrastructure provider controller.
+
+Also, check to see whether different versions were used in the initial cluster
+creation for the control plane and the worker nodes, as Kubernetes allows a
+limited [version skew](https://kubernetes.io/releases/version-skew-policy/).
+
+### MachineDeployment does not function
```

Comment:
"MachineDeployment with Docker Provider does not function" or smth. This item is specific to Docker Provider

Reply:
Added 'with Docker Provider'.